manifests/install.pp

```diff
@@ -25,10 +26,23 @@
         }
       }
 
-      download_file { 'NSCP-Installer':
-        url                   => $source,
-        destination_directory => $nsclient::download_destination,
-        require               => File[$nsclient::download_destination],
+      if $nsclient::proxy_url {
```

Comment:
this if/else block isn't needed. You can always pass the `$nsclient::proxy_url` variable. It won't do anything if it is undef.

Reply:
Ok I wasn't sure. Will check this.